Sani_Zayyad/lab3/Source/main.cpp

```diff
@@ -0,0 +1,205 @@
+#include "stack.hpp"
+#include <iostream>
+#include <fstream>
+#include <string>
+
+void ReadFromFile(std::string filename);
+std::string getInfix(std::string input);
+Data makeInfix(char sign, Data firstArg, Data secondArg);
+bool isAlpha(const char ch);
+bool isDigit(const char ch);
+bool isOperator(const char ch);
+
+
+void ReadFromFile(std::string filename)
+{
+    std::ifstream file(filename);
+
+    if (file.is_open())
+    {
+        int count = 0;
+        while (!file.eof())
+        {
+            count++;
+            std::string input;
+            getline(file, input);
+            std::cout << "test #" << count << " \"" + input + "\""<< "\n";
+            std::string output = getInfix(input);
+            std::cout<< "---->  FINAL EXPRESSION : "<< output <<"\n\n";
+        }
+    }
+    else
+    {
+        std::cout << "File not opened"<<"\n";
+    }
+}
+
+std::string getInfix(std::string postfixExpression){
+    Stack stack;
+    for (auto i = postfixExpression.cbegin(); i < postfixExpression.end(); ++i) {
+       char elem = *i;
```

Comment:
Это стоит заменить на ranged-based for

Reply:
I can't use ranged-base for loop , because range for loop has a limitation ,you can’t add a condition to it and you can't change the iteration . in my code there's a while loop where i iterated through the remaining element ,which is not possible with range for...

Я не могу использовать ranged-base for loop, потому что range for loop имеет ограничения, вы не можете добавить к нему условие и не можете изменить итерацию . в моем коде есть цикл while, где я перебирал оставшийся элемент ,что невозможно с помощью range-based for ..  
 :)